### DISCUSSION TITLE
2.2.18: arguments over-quoted with nested referenced scripts?

### DISCUSSION THREAD

**Discussion by lkraav** [composer/composer#11255] [Q&A / Support]
```
$ ./composer --version
Composer 2.2.18 2022-08-20 11:33:38
```

Looking to build something like this

```json
"scripts": [
  "db-export": [
    "@db-export:global",
    "@db-export:no-data",
    ...
  ],
  "db-export-cmd": "wp db export - --skip-extended-insert --skip-comments",
  "db-export-global": "@db-export-cmd --exclude_tables=wp_ac_segments > wp.sql",
  "db-export-no-data": "@db-export-cmd --no-data=true --tables='wp_ac_segments' >> wp.sql"
]
```

Now we fail with "Too many positional arguments", because nested scripts args are quoted each:

```
$ ./composer db-export
> wp db export --skip-extended-insert --skip-comments - '--exclude_tables=wp_ac_segments' '>' 'wp.sql'
Error: Too many positional arguments: > wp.sql
```

If I repeat "db-export-cmd" contents everywhere, it works.

Is it possible to make it work as shown above?

**Comment by Seldaek** [comment]
hmm yeah those extra args aren't really meant to be used as a way to do bash aliases etc like you're doing here.. You can try with Composer 2.5 tho we did fix and update a few things there since 2.2, but it may still not work I am not entirely sure it's kinda on the edge of things we want to support IMO.

**Reply by lkraav** [reply]
Not sure, from the looks of it I'm not doing anything super-magical?

Not able to try 2.5 immediately due to dependencies not allowing, but I'll report back when I have more intel.
